eureka-core/src/main/java/com/netflix/eureka/util/StatusUtil.java

```diff
@@ -50,11 +52,16 @@ public StatusInfo getStatusInfo() {
         builder.add("registered-replicas", replicaHostNames.toString());
         builder.add("available-replicas", upReplicas.toString());
         builder.add("unavailable-replicas", downReplicas.toString());
+        
+        // Only set the healthy flag if a threshold has been configured.
+        if (peerEurekaNodes.getMinNumberOfAvailablePeers() > 0) {
```

Comment:
What about a not configured default of -1? Technically you can run with a single eureka server (obv not recommended), in which case you would legitimately have 0 peers.